### PR TITLE
[REF] Fold getACLs back into the calling function

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
[REF] Fold getACLs back into the calling function

Before
----------------------------------------
Complex code flow

After
----------------------------------------
Suddenly it's obvious the plot twist that is iminent on getACLRoles....

Technical Details
----------------------------------------
Not called in the civi universe outside of core (it's protected anyway)

Comments
----------------------------------------

### CODE REVIEW

File: CRM/ACL/BAO/ACL.php [hunk 1]
```diff
@@ -258,7 +226,18 @@ public static function getAllByContact($contact_id) {
 
     /* First, the contact-specific ACLs, including ACL Roles */
     if ($contact_id) {
-      $result += self::getACLs((int) $contact_id);
+      $query = " SELECT acl.*
+      FROM civicrm_acl acl
+      WHERE   acl.entity_table   = 'civicrm_contact'
```
Comment (seamuslee001) [inline CRM/ACL/BAO/ACL.php:231]: how does this even work?

Comment (eileenmcnaughton) [reply CRM/ACL/BAO/ACL.php:231]: @seamuslee001 in terms of 'when would entity_table be equal to 'civicrm_contact'? - my suspicion is that it has not been possible for a decade - I'm still trying to test that theory

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
unrelated fail but re-running 


api_v3_SyntaxConformanceTest::testCreateSingleValueAlter with data set #82 ('UFField')
Failure in api call for UFField create:  Undefined index: type
#0 /home/jenkins/bknix-dfl/build/core-19872-8p1uw/web/sites/all/modules/civicrm/CRM/Utils/Date.php(1953): PHPUnit\Util\ErrorHandler::handleError(8, 'Undefined index...', '/home/jenkins/b...', 1953, Array)
#1 /home/jenkins/bknix-dfl/build/core-19872-8p1uw/web/sites/all/modules/civicrm/CRM/Core/BAO/UFGroup.php(510): CRM_Utils_Date::addDateMetadataToField(Array, Array)
#2 /home/jenkins/bknix-dfl/build/core-19872-8p1uw/web/sites/all/modules/civicrm/CRM/Core/BAO/UFGroup.php(346): CRM_Core_BAO_UFGroup::formatUFField(Object(CRM_Core_DAO), Object(CRM_Core_DAO), Array, Array, Array, 4)
#3 /home/jenkins/bknix-dfl/build/core-19872-8p1uw/web/sites/all/modules/civicrm/CRM/Core/BAO/UFGroup.php(2910): CRM_Core_BAO_UFGroup::getFields(118, false, NULL, NULL, NULL, true, NULL, true)
#4 /home/jenkins/bknix-dfl/build/core-19872-8p1uw/web/sites/all/modules/civicrm/CRM/Core/BAO/UFField.php(107): CRM_Core_BAO_UFGroup::calculateGroupType('118', true)
#5 /home/jenkins/bknix-dfl/build/core-19872-8p1uw/web/sites/all/modules/civicrm/api/v3/utils.php(1298): CRM_Core_BAO_UFField::create(Array, Array)
#6 /home/jenkins/bknix-dfl/build/core-19872-8p1uw/web/sites/all/modules/civicrm/api/v3/UFField.php(30): _civicrm_api3_basic_create('CRM_Core_BAO_UF...', Array, 'UFField')

**Comment by eileenmcnaughton** [comment]
@seamuslee001 this has passed now - it just moves code around (the questions it raises around the 'civicrm_contact' in entity_value I hope to get to

**Pull request merged**